lib/src/util/__tests__/vulnerabilitySeverityRatings.test.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import {
+  formatVulnerabilityScore,
+  getVulnerabilitySeverityRating
+} from '../vulnerabilitySeverityRatings';
+
+describe('vulnerabilitySeverityLevels utilities', function() {
+  const errorRange = 'Vulnerability score must be between 0.0 and 10.0.';
+  const errorDecimal = 'Vulnerability score must be a valid number with one decimal place.';
+
+  describe('formatVulnerabilityScore', function() {
+    it('returns the correct Vulnerability (CVSS) score string given floating point number', function() {
+      expect(() => formatVulnerabilityScore(-1)).toThrowError(errorRange);
+      expect(() => formatVulnerabilityScore(11)).toThrowError(errorRange);
+
+      expect(() => formatVulnerabilityScore(NaN)).toThrowError(errorDecimal);
+      expect(() => formatVulnerabilityScore(-Infinity)).toThrowError(errorRange);
+      expect(() => formatVulnerabilityScore(Infinity)).toThrowError(errorRange);
+
+      expect(() => formatVulnerabilityScore(7.00000000002)).toThrowError(errorDecimal);
+      expect(() => formatVulnerabilityScore(4.55)).toThrowError(errorDecimal);
+      expect(() => formatVulnerabilityScore(5.88)).toThrowError(errorDecimal);
+
+      expect(formatVulnerabilityScore(-0)).toBe('0.0');
+      expect(formatVulnerabilityScore(0)).toBe('0.0');
+      expect(formatVulnerabilityScore(2)).toBe('2.0');
+      expect(formatVulnerabilityScore(3.3)).toBe('3.3');
+      expect(formatVulnerabilityScore(10)).toBe('10.0');
+    });
+  });
+
+  describe('getVulnerabilitySeverityRating', function() {
+    it('returns the correct severity rating for each valid vulnerability score', function() {
+      expect(getVulnerabilitySeverityRating('0.0')).toBe('none');
+      expect(getVulnerabilitySeverityRating('0.1')).toBe('low');
+      expect(getVulnerabilitySeverityRating('3.9')).toBe('low');
+      expect(getVulnerabilitySeverityRating('4.0')).toBe('medium');
+      expect(getVulnerabilitySeverityRating('6.9')).toBe('medium');
+      expect(getVulnerabilitySeverityRating('7.0')).toBe('high');
+      expect(getVulnerabilitySeverityRating('8.9')).toBe('high');
+      expect(getVulnerabilitySeverityRating('9.0')).toBe('critical');
+      expect(getVulnerabilitySeverityRating('10.0')).toBe('critical');
+
+      expect(getVulnerabilitySeverityRating(7.3)).toBe('high');
+      expect(getVulnerabilitySeverityRating(10)).toBe('critical');
+
+      expect(() => getVulnerabilitySeverityRating(1.00000000001)).toThrowError(errorDecimal);
+      expect(() => getVulnerabilitySeverityRating('11.0' as any)).toThrow(TypeError);
```

Comment:
Test out of range numbers and exotic numbers too.

Reply:
> Test out of range numbers and exotic numbers too.

Done as part of: [9da14ac](https://github.com/sonatype/sonatype-react-shared-components/pull/1202/commits/9da14acabcee920848ac7dcdfa17e2a582c7a715)